operator/src/test/java/org/keycloak/operator/testsuite/integration/KeycloakIngressTest.java

```diff
@@ -188,4 +193,81 @@ public void testMainIngressDurability() {
                     assertThat(k8sclient.network().v1().ingresses().inNamespace(namespace).list().getItems().size()).isEqualTo(0);
                 });
     }
+
+    @Test
+    public void testCustomIngressDeletion() {
+
+        Keycloak defaultKeycloakDeployment = K8sUtils.getDefaultKeycloakDeployment();
+        String kcDeploymentName = defaultKeycloakDeployment.getMetadata().getName();
+        Resource<Ingress> customIngressDeployedManuallySelector = null;
+        Ingress customIngressCreatedManually;
+
+        try {
+
+            customIngressCreatedManually = createCustomIngress(kcDeploymentName, namespace, 8443);
+
+            customIngressDeployedManuallySelector = k8sclient
+                    .network()
+                    .v1()
+                    .ingresses()
+                    .inNamespace(namespace)
+                    .withName(customIngressCreatedManually.getMetadata().getName());
+
+            Awaitility.await().atMost(1, MINUTES).untilAsserted(() -> {
+                assertThat(k8sclient.network().v1().ingresses().inNamespace(namespace).list().getItems().size()).isEqualTo(1);
+            });
+
+            Log.info("Redeploying the Keycloak CR with default Ingress disabled");
+            defaultKeycloakDeployment = K8sUtils.getDefaultKeycloakDeployment();
+            defaultKeycloakDeployment.getSpec().setIngressSpec(new IngressSpec());
+            defaultKeycloakDeployment.getSpec().getIngressSpec().setIngressEnabled(false);
+
+            K8sUtils.deployKeycloak(k8sclient, defaultKeycloakDeployment, true);
+
+            Awaitility.await().untilAsserted(() -> {
+                Log.info("Make sure the Custom Ingress still remains");
+                assertThat(k8sclient.network().v1().ingresses().inNamespace(namespace).list().getItems().size()).isEqualTo(1);
+            });
+
+        } catch (Exception e) {
+            savePodLogs();
+            throw e;
+        } finally {
+            Log.info("Destroying the Custom Ingress created manually to avoid errors in others Tests methods");
+            assertThat(customIngressDeployedManuallySelector).isNotNull();
```

Comment:
Did you mean this?
```suggestion
            assertThat(customIngressDeployedManuallySelector.get()).isNotNull();
```

Reply:
Actually, nevermind. There an `if` for null-check instead. Otherwise the following line will fail if `customIngressDeployedManuallySelector == null`.

Reply:
Done!

Thanks.